netty-reactive-streams/src/main/java/com/typesafe/netty/HandlerPublisher.java

```diff
@@ -203,32 +203,40 @@ private void provideSubscriber(Subscriber<? super T> subscriber) {
 
     @Override
     public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
-        verifyRegisteredWithRightExecutor(ctx);
+        // If the channel is not yet registered, then it's not safe to invoke any methods on it, eg read() or close()
+        // So don't provide the context until it is registered.
+        if (ctx.channel().isRegistered()) {
+            provideChannelContext(ctx);
+        }
+    }
+
+    @Override
+    public void channelRegistered(ChannelHandlerContext ctx) throws Exception {
+        provideChannelContext(ctx);
+        ctx.fireChannelRegistered();
```

Comment:
There's no guarantee that it's the last one - for one, there may be a subscriber after it, but even then, HandlerPublishers only publish the types of messages they are configured to handle, the rest they forward on.  You could for example have two HandlerPublishers in the pipeline publishing two different types of messages - though they would interfere with each others backpressure if you did that.